src/common/src/catalog/schema.rs

```diff
@@ -197,6 +197,21 @@ impl Schema {
             true
         }
     }
+
+    /// Check if the schema is simaler to the iceberg table schema.
+    pub fn has_simaler_schema(&self, arrow_schema: &ArrowSchema) -> bool {
+        if self.fields.len() != arrow_schema.fields().len() {
+            return false;
+        }
+        self.fields
+            .iter()
+            .zip_eq_fast(arrow_schema.fields())
+            .all(|(field, arrow_field)| {
+                field
```

Comment:
We only check that the type in schema is simalar and ignore the other field (such as field name). Do we need to check those?🤔 cc @liurenjie1024

Reply:
You can provide a parameter in this function to users to choose whether check names or not.